src/cloudfront-auth.ts

```diff
@@ -30,7 +30,11 @@ export interface CloudFrontAuthProps {
    * @default - a new client will be generated
    */
   client?: cognito.UserPoolClient
-  userPool: cognito.IUserPool
+  userPool?: cognito.IUserPool
+  /**
+   * Custom domain for the Cognito User Pool.
+   */
+  userPoolDomain?: string
```

Comment:
don't forget to update the docs around "Cognito domain can be automatically retrieved if it's not known"